src/components/tipRecords/SoundCloudPlayer.vue

```diff
@@ -0,0 +1,224 @@
+<template>
+  <div
+    class="soundcloud-player"
+    @click.stop
+  >
+    <iframe
+      :id="`sc-player-${tip.id}`"
+      class="soundcloud-iframe"
+      :src="playUrl"
+    />
+    <div
+      class="play"
+    >
+      <div
+        v-if="loading"
+        class="spinner-border spinner-border-sm text-primary loading"
+      />
+      <img
+        v-else-if="isPlaying"
+        class="pause-button"
+        src="../../assets/buttonPause.svg"
+        @click="togglePlay"
+      >
+      <img
+        v-else-if="!isPlaying"
+        class="play-button"
+        src="../../assets/buttonPlay.svg"
+        @click="togglePlay"
+      >
+    </div>
+    <div
+      ref="soundWave"
+      class="sound-wave"
+      @click="mousePosition"
+    >
+      <div class="wave-wrapper">
+        <img
+          class="wave-grey"
+          src="../../assets/soundWave.svg"
+        >
+        <div
+          class="wave-green"
+          :style="{ 'width': `${position}%`, 'background-size': `${1 / position * 10000}% 100%` }"
+        />
+      </div>
+      <div class="wave-wrapper flipped">
+        <img
+          class="wave-grey"
+          src="../../assets/soundWave.svg"
+        >
+        <div
+          class="wave-green"
+          :style="{ 'width': `${position}%`, 'background-size': `${1 / position * 10000}% 100%` }"
+        />
+      </div>
+      <div class="display-time">
+        <span>{{ displayProgress }}</span>
+        <span class="right">{{ displayDuration }}</span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import moment from 'moment';
+
+export default {
+  props: {
+    tip: { type: Object, required: true },
+    scApiUrl: { type: String, required: true },
+  },
+  data() {
+    return {
+      player: null,
+      isPlaying: false,
+      loading: true,
+      progress: 0,
+      duration: 0,
+      position: 0,
+    };
+  },
+  computed: {
+    playUrl() {
+      return `${this.scApiUrl}?url=${this.tip.url}`;
+    },
+    displayProgress() {
+      return moment.utc(this.progress).format(this.progress < 3600000 ? 'mm:ss' : 'h:mm:ss');
+    },
+    displayDuration() {
+      return moment.utc(this.duration).format(this.duration < 3600000 ? 'mm:ss' : 'h:mm:ss');
+    },
+  },
```

Comment:
`moment` the way how it is imported consumes significant part of the bundle:
https://feature-rich-content-preview.aggregator.z52da5wt.xyz/report.html
https://develop.aggregator.z52da5wt.xyz/report.html

I propose to implement this function by ourselves or to use a Date object:
```suggestion
  },
  filters: {
    formatTime: timestamp => {
      const seconds = Math.floor(timestamp / 1000);
      const minutes = Math.floor(seconds / 60);
      const hours = Math.floor(minutes / 60);
      return [
        ...hours ? [hours] : [],
        ...[minutes % 60, seconds % 60].map(t => t.toString().padStart(2, '0')),
      ].join(':');
    },
    // or using a Date object:    
    formatTime: timestamp => new Date(timestamp).toLocaleTimeString([], {
      ...timestamp > 60 * 60 * 1000  ? { hour: 'numeric' },
      minute: '2-digit',
      second: '2-digit',
      timeZone: 'UTC',
    }),
  },
```

Both `Date` and `moment` are not handling the big values properly. For example,
```
> moment.utc(100000000).format('h:mm:ss')
3:46:40 // instead of 27:46:40
```
I'm not sure is it the case of SoundCloud, so I'm not sure which implementation is better.